File: year2020/month12/day1217/ms_1217.cpp
#include <iostream>
#include <cmath>

using namespace std;

long long solution(long long n);

int main() {
	long long n;

	cin >> n;

	if (1 <= n && n <= 50000000000000) {
		cout << solution(n) << endl;
	}
}

long long solution(long long n) {
	long long answer = -1;
	long long value = floor(sqrt(n));

	if (value * value == n) {
		answer = (value + 1) * (value + 1);
	}

	return answer;
}

File: year2020/month12/day1217/ms_1217.cs
using System;

namespace algorithm_124
{
    class ms_1217
    {
        static long solution(long n) 
        {
            long answer = -1;
            long sqrt = Convert.ToInt64(Math.Truncate(Math.Sqrt(n)));
            
            if(sqrt * sqrt == n) 
            {
                answer = (sqrt + 1) * (sqrt + 1);
            }
            
            return answer;
        }

        static void Main(string[] args)
        {
            long n = Convert.ToInt64(Console.ReadLine());

            if(1 <= n && n <= 50000000000000) 
            {
                Console.WriteLine(solution(n));
            }
        }
    }
}
